ocs_ci/ocs/resources/pod.py

```diff
@@ -1727,12 +1727,16 @@ def get_osd_removal_pod_name(osd_id, timeout=60):
 
     """
     ocs_version = config.ENV_DATA["ocs_version"]
-    if Version.coerce(ocs_version) == Version.coerce("4.7"):
+    if Version.coerce(ocs_version) <= Version.coerce("4.6"):
+        pattern = f"ocs-osd-removal-{osd_id}"
+    elif Version.coerce(ocs_version) == Version.coerce("4.7"):
         pattern = "ocs-osd-removal-job"
     elif Version.coerce(ocs_version) == Version.coerce("4.8"):
         pattern = "ocs-osd-removal-"
+    elif Version.coerce(ocs_version) >= Version.coerce("4.9"):
+        pattern = "ocs-osd-removal-job"
     else:
-        pattern = f"ocs-osd-removal-{osd_id}"
+        logger.info(f"Didn't find ocs version {ocs_version}")
```

Comment:
you can work with dictionary:
for example:
```
dic={"4.6":f"ocs-osd-removal-{osd_id}",
"4.7":"ocs-osd-removal-job"
}

ocs_version = config.ENV_DATA["ocs_version"]
pattern=dic[ocs_version]
```

Reply:
Yes, good idea.

Reply:
There is no `switch case`  in python:)